### ISSUE TITLE
Remote doesn't seem to work with Firefox Windows/Linux

### ISSUE THREAD

**Issue by laurinneff** [deckgo/deckdeckgo#362]
I'm new to DeckDeckGo and still figuring things out, so it might be a user issue, but whenever I try to connect using my phone, it doesn't continue past "Connected with the signaling server, waiting for the presentation..."

**Comment by peterpeterparker** [comment]
@laurinneff thx for reaching out. I just gave a try on two devices and all seems up and running aka I was able to establish a connection.

Could you give a try refreshing the presentation (browser refresh) or turning off/on the presentation to refresh the connection, that should made the trick.

Let me know if this worked out.

<img width="1170" alt="Capture d’écran 2019-09-25 à 20 22 56" src="https://user-images.githubusercontent.com/16886711/65628777-6f99fe00-dfd2-11e9-9c53-6275096e023e.png">

**Comment by laurinneff** [comment]
I tried it on my phone (Android, tried Firefox and Chrome), my iPad (Chrome), my PC (Firefox) and it didn't work on any of these device/browser combos

**Comment by laurinneff** [comment]
I also tried refreshing the presentation and the remote control

**Comment by peterpeterparker** [comment]
Same here, different browsers, no problem :(

Are you on different network maybe or same network (for the remote and the presentation)?

**Comment by laurinneff** [comment]
I tried both WiFi/mobile data (on phone), the network was the same.

**Comment by peterpeterparker** [comment]
Mmmh weird, I even do see other people than me having a successful connection between deck and remote

Could you open a presentation on one of your devices, keep the browser, and send me a screenshot as the one I provided above?

Then I'll try to connect and see if I see something weird

**Comment by laurinneff** [comment]
![screenshot](https://user-images.githubusercontent.com/29987831/65629773-4d08e480-dfd4-11e9-94d3-195dea13d8eb.png)

**Comment by peterpeterparker** [comment]
I couldn't neither. It was hanging on the webrtc signaling server so I restarted the turnserver

Could you give a try again to refresh your presentation, check is number and try to connect?
Does that work now that I've restarted that server?

**Comment by laurinneff** [comment]
I still get the same issue.

I'm now trying to run the presentation on a different device and OS to see if that's the issue (Manjaro Linux on a laptop instead of Windows on a custom built desktop)

**Comment by laurinneff** [comment]
The different device/OS didn't work either.

**Comment by peterpeterparker** [comment]
Crazy.

What's the msg, "CONNECTED WITH THE SIGNALING SERVER, WAITING FOR THE PRESENTATION..." or another one?

**Comment by laurinneff** [comment]
Yes, it's the connected one

**Comment by laurinneff** [comment]
I just tried chrome on said laptop (as I can see that you're using chrome on your mac(book)), and there it works. Seems to be an issue with firefox

**Comment by laurinneff** [comment]
Since it seems to be a firefox issue, here's the console output on firefox (from reloading to attempting to connect):
```
here 
ResizeObserver {  }
main.02596b182fb909c7a6b3.js:16:278523
Apply() called. customcss.js:29:10
Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://deckdeckgo.com/assets/icons/ionicons/clipboard.svg. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt).
TypeError: NetworkError when attempting to fetch resource.
Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://deckdeckgo.com/assets/icons/ionicons/share.svg. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt).
TypeError: NetworkError when attempting to fetch resource.
Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://deckdeckgo.com/assets/icons/deckdeckgo.svg. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt).
TypeError: NetworkError when attempting to fetch resource.
Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://deckdeckgo.com/assets/icons/ionicons/phone-portrait.svg. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt).
TypeError: NetworkError when attempting to fetch resource.
Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://deckdeckgo.com/assets/icons/ionicons/close.svg. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt).
TypeError: NetworkError when attempting to fetch resource.
ICE failed, add a STUN server and see about:webrtc for more details
```

**Comment by peterpeterparker** [comment]
What Firefox version?

I just tried with v69, all good

<img width="996" alt="Capture d’écran 2019-09-25 à 20 53 46" src="https://user-images.githubusercontent.com/16886711/65630917-abcf5d80-dfd6-11e9-906d-df5d487df98e.png">

**Comment by laurinneff** [comment]
Also, here's the .har file produced by firefox's network analysis if that helps you: https://1drv.ms/u/s!AsIrheqdsn_Xgr8R_Dw9hMcpa9yXDg?e=tpsafr

**Comment by laurinneff** [comment]
I'm on the same version. Maybe it works on Firefox/macOS but not on Firefox/Windows or Firefox/Linux

**Comment by peterpeterparker** [comment]
Maybe, I've restarted both turn server and connecting server, so there should be no blocker on these and the signal should be able to go through.

I've tested it, on Mac (Firefox, Safari and Chrome) and on my Android phone, all good.

So yep, maybe something weird with Firefox on Windows/Linux.

I'll try tomorrow with another more restrictive network to be sure that isn't an access problem.

But if not, not much I could do unfortunately. It would mean, like you said, that there is something weird with such browser on such devices regarding the standard WebRTC

Agree with that?

P.S. If interested, for example, the [remote](https://github.com/deckgo/deckdeckgo/blob/master/webcomponents/remote/src/services/communication/communication.service.tsx)) service with establish the connection on the presentation side

**Comment by laurinneff** [comment]
Hope you can fix it, otherwise I'll have to use chrome, which wouldn't really be a deal breaker but it'd be nice if I could use my favorite browser. 

Also, unrelated question but what defines if the remote is light theme or dark theme? Because on Firefox on my PC it's dark, but on my phone it's light

**Comment by peterpeterparker** [comment]
Let's see tomorrow when I try on other network but I fear that there isn't much I could do on our side. Let's see, I'll keep you posted.

The dark/light them are defined by the user preferences of his/her OS (see CSS https://developer.mozilla.org/fr/docs/Web/CSS/@media/prefers-color-scheme)

@grantlouisherman in https://github.com/deckgo/deckdeckgo/issues/227 is currently experimenting a POC in order to implement a dark/light them switcher which we would ultimately implement in all our apps

**Comment by laurinneff** [comment]
Thanks for the reply, a theme switcher would be nice since AOSP only has system-wide dark theme since 10, and I still have 9

**Comment by peterpeterparker** [comment]
Agree, would definitely be cool!

Thx for the feedbacks and for the tests!

**Comment by peterpeterparker** [comment]
@laurinneff as discussed yesterday, I just tried to run the presentation with Firefox on a more restrictive network and used my Android phone on another network (mobile) and I didn't faced any problem, I was able to establish the connection and remote control my presentation.

Therefore, unfortunately, I think these isn't much I could do. This validate that the servers (signaling and turn) are working correctly. Moreover, as pointed above, the implementation is based on a standard.

Maybe like you said, something weird with Firefox on Linux or Windows, hopefully this will be fixed some day 🤞

Thx again for all tests and for giving a try to DeckDeckGo!
